state/protocol/events/gadgets/views.go

```diff
@@ -0,0 +1,61 @@
+package gadgets
+
+import (
+	"sync"
+
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/state/protocol/events"
+)
+
+// Views is a protocol events consumer that provides an interface to
+// subscribe to callbacks when chain state reaches a particular view.
+type Views struct {
```

Comment:
Good flag. We should not have this issue here because we are using finalization events to trigger these callbacks. All honest nodes will finalize the same blocks in the same order. "The first finalized block with view >= `v`" is a deterministically computable block and doesn't depend on the order the node sees blocks.